app/src/main/java/com/alexbar/layoutbasic/movies_app/model/Media.kt

```diff
@@ -0,0 +1,45 @@
+package com.alexbar.layoutbasic.movies_app.model
+
+import android.annotation.SuppressLint
+import com.alexbar.layoutbasic.movies_app.utils.MovieConstants
+import com.google.gson.annotations.SerializedName
+import java.text.SimpleDateFormat
+
+data class Media(
+    @SerializedName("title")
+    val title: String? = null,
+    @SerializedName("name")
+    val name: String? = null,
+    @SerializedName("overview")
+    val description: String,
+    @SerializedName("poster_path")
+    val imagePosterUrl: String,
+    @SerializedName("backdrop_path")
+    val imageBackdropUrl: String,
+    @SerializedName("media_type")
+    val mediaType: String,
+    @SerializedName("first_air_date")
+    val firstAirDate: String?,
+    @SerializedName("release_date")
+    val releaseDate: String?,
+    @SerializedName("vote_average")
+    val voteAverage: Double
+) {
+    fun getImagePosterCompleteUrl() = "${MovieConstants.base_url}$imagePosterUrl"
+    fun getImageBackdropCompleteUrl() = "${MovieConstants.base_url}$imageBackdropUrl"
+    fun getVoteAverageFormatted() = String.format("%.1f", voteAverage)
+
+    @SuppressLint("SimpleDateFormat")
+    fun getAirDateFormatted(): String {
+        val originalFormat = SimpleDateFormat("yyyy-MM-dd")
+        val newFormat = SimpleDateFormat("MMMM dd, yyyy")
+        return originalFormat.parse(firstAirDate!!)?.let { newFormat.format(it) } ?: ""
+    }
+
+    @SuppressLint("SimpleDateFormat")
+    fun getReleaseDateFormatted(): String {
+        val originalFormat = SimpleDateFormat("yyyy-MM-dd")
+        val newFormat = SimpleDateFormat("MMMM dd, yyyy")
+        return originalFormat.parse(releaseDate!!)?.let { newFormat.format(it) } ?: ""
+    }
+}
```

Comment:
@ezatpanah  Is it a good practice to have all this logic inside the data class, or should I move it elsewhere, such as to a utils file? And regarding formatting dates, the date returned by the API is like "2024-02-02" and the one I want to display in the UI is like "February 02, 2024." Is the way I implemented it correct, or is there a more optimal way to do it?

Reply:
Separating formatting logic from data classes is a good practice. It makes your code more maintainable. You can use utility or extension functions for formatting dates and URLs. so my suggestion here is . first separate your data class and have it in one dataclass file like :
```
data class Media(
    @SerializedName("title")
    val title: String? = null,
    @SerializedName("name")
    val name: String? = null,
    @SerializedName("overview")
    val description: String,
    @SerializedName("poster_path")
    val imagePosterUrl: String,
    @SerializedName("backdrop_path")
    val imageBackdropUrl: String,
    @SerializedName("media_type")
    val mediaType: String,
    @SerializedName("first_air_date")
    val firstAirDate: String?,
    @SerializedName("release_date")
    val releaseDate: String?,
    @SerializedName("vote_average")
    val voteAverage: Double
)
```
and then :
then for following the SOLID principles, particularly the Single Responsibility Principle (SRP), suggests that each class should have only one reason to change. In the case of the data class Media, its responsibility is to hold data related to media items. Therefore, it's not ideal to burden it with responsibilities like formatting dates or ... . so as I mentioned before , you can put your method in another kotlin file as extension func like : 
First, create a file named MediaExtensions.kt:

```
fun Media.getImagePosterCompleteUrl() = "${MovieConstants.base_url}${this.imagePosterUrl}"
fun Media.getImageBackdropCompleteUrl() = "${MovieConstants.base_url}${this.imageBackdropUrl}"
fun Media.getVoteAverageFormatted() = String.format("%.1f", this.voteAverage)

@SuppressLint("SimpleDateFormat")
fun String?.formatDate(): String {
    if (this.isNullOrEmpty()) return ""
    val originalFormat = SimpleDateFormat("yyyy-MM-dd")
    val newFormat = SimpleDateFormat("MMMM dd, yyyy")
    return try {
        val date = originalFormat.parse(this)
        newFormat.format(date)
    } catch (e: ParseException) {
        ""
    }
}
```
With this organization, the extension functions remain tied to the Media class and String class respectively, but they are kept in separate files, adhering to the SRP. This keeps your code clean and modular, making it easier to manage and maintain. :)